### ISSUE TITLE
read_sql_query() fails when using nullable integer dtype

### ISSUE THREAD

**Issue by Nihhaar** [dask/dask#6010]
Minimal bug report:
```python
from dask import dataframe as ddf
import pandas as pd
import numpy as np
import sqlite3
from sqlalchemy import sql, types, cast

# Generate test data
db_file = "testdb.db"
conn_str = "sqlite:///{}".format(db_file)
table_name = "test"

id = ['0', '1', '2', '3', '4']
v1 = [0, 1, None, 2, 3]
v2 = [0, 1, 2, 3, 4]

pd.DataFrame({"id": str(id), "v1": v1, "v2": v2}).to_sql(table_name,
                                                    sqlite3.connect(db_file),
                                                    if_exists="replace")

# Reproduce problem
columns = {"id": 'int', 'v1': 'Int64', 'v2': 'int'}
meta = pd.DataFrame({name: pd.Series(dtype=t) for name, t in columns.items()})
meta = pd.DataFrame(meta)[0:0]

x = ddf.read_sql_table(table_name, conn_str,
                       index_col=sql.cast(sql.column("id"), types.BigInteger).label("id"),
                       meta=meta,
                       columns=list(meta.columns),
                       npartitions=10)

x.reset_index() # Error here
```

Produces error:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/workspace/lib/python3.6/site-packages/dask/dataframe/utils.py in raise_on_meta_error(funcname, udf)
    167     try:
--> 168         yield
    169     except Exception as e:

~/workspace/lib/python3.6/site-packages/dask/dataframe/core.py in _emulate(func, *args, **kwargs)
   4724     with raise_on_meta_error(funcname(func), udf=kwargs.pop("udf", False)):
-> 4725         return func(*_extract_meta(args, True), **_extract_meta(kwargs, True))
   4726

~/workspace/lib/python3.6/site-packages/dask/dataframe/core.py in _extract_meta(x, nonempty)
   4704     elif isinstance(x, tuple):
-> 4705         return tuple([_extract_meta(_x, nonempty) for _x in x])
   4706     elif isinstance(x, dict):

~/workspace/lib/python3.6/site-packages/dask/dataframe/core.py in <listcomp>(.0)
   4704     elif isinstance(x, tuple):
-> 4705         return tuple([_extract_meta(_x, nonempty) for _x in x])
   4706     elif isinstance(x, dict):

~/workspace/lib/python3.6/site-packages/dask/dataframe/core.py in _extract_meta(x, nonempty)
   4700     if isinstance(x, (Scalar, _Frame)):
-> 4701         return x._meta_nonempty if nonempty else x._meta
   4702     elif isinstance(x, list):

~/workspace/lib/python3.6/site-packages/dask/dataframe/core.py in _meta_nonempty(self)
    332         """ A non-empty version of `_meta` with fake data."""
--> 333         return meta_nonempty(self._meta)
    334

~/workspace/lib/python3.6/site-packages/dask/utils.py in __call__(self, arg, *args, **kwargs)
    504         meth = self.dispatch(type(arg))
--> 505         return meth(arg, *args, **kwargs)
    506

~/workspace/lib/python3.6/site-packages/dask/dataframe/utils.py in meta_nonempty_dataframe(x)
    384     idx = meta_nonempty(x.index)
--> 385     data = {i: _nonempty_series(x.iloc[:, i], idx=idx) for i, c in enumerate(x.columns)}
    386     res = pd.DataFrame(data, index=idx, columns=np.arange(len(x.columns)))

~/workspace/lib/python3.6/site-packages/dask/dataframe/utils.py in <dictcomp>(.0)
    384     idx = meta_nonempty(x.index)
--> 385     data = {i: _nonempty_series(x.iloc[:, i], idx=idx) for i, c in enumerate(x.columns)}
    386     res = pd.DataFrame(data, index=idx, columns=np.arange(len(x.columns)))

~/workspace/lib/python3.6/site-packages/dask/dataframe/utils.py in _nonempty_series(s, idx)
    565         entry = _scalar_from_dtype(dtype)
--> 566         data = np.array([entry, entry], dtype=dtype)
    567

TypeError: data type not understood

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-23-b66fb60ba4c5> in <module>
----> 1 x.reset_index()

~/workspace/lib/python3.6/site-packages/dask/dataframe/core.py in reset_index(self, drop)
    447             Do not try to insert index into dataframe columns.
    448         """
--> 449         return self.map_partitions(M.reset_index, drop=drop).clear_divisions()
    450
    451     @property

~/workspace/lib/python3.6/site-packages/dask/dataframe/core.py in map_partitions(self, func, *args, **kwargs)
    609         >>> ddf.map_partitions(func).clear_divisions()  # doctest: +SKIP
    610         """
--> 611         return map_partitions(func, self, *args, **kwargs)
    612
    613     @insert_meta_param_description(pad=12)

~/workspace/lib/python3.6/site-packages/dask/dataframe/core.py in map_partitions(func, *args, **kwargs)
   4764         # Use non-normalized kwargs here, as we want the real values (not
   4765         # delayed values)
-> 4766         meta = _emulate(func, *args, udf=True, **kwargs)
   4767     else:
   4768         meta = make_meta(meta, index=meta_index)

~/workspace/lib/python3.6/site-packages/dask/dataframe/core.py in _emulate(func, *args, **kwargs)
   4723     """
   4724     with raise_on_meta_error(funcname(func), udf=kwargs.pop("udf", False)):
-> 4725         return func(*_extract_meta(args, True), **_extract_meta(kwargs, True))
   4726
   4727

/usr/lib/python3.6/contextlib.py in __exit__(self, type, value, traceback)
     97                 value = type()
     98             try:
---> 99                 self.gen.throw(type, value, traceback)
    100             except StopIteration as exc:
    101                 # Suppress StopIteration *unless* it's the same exception that

~/workspace/lib/python3.6/site-packages/dask/dataframe/utils.py in raise_on_meta_error(funcname, udf)
    187         )
    188         msg = msg.format(" in `{0}`".format(funcname) if funcname else "", repr(e), tb)
--> 189         raise ValueError(msg)
    190
    191

ValueError: Metadata inference failed in `reset_index`.

You have supplied a custom function and Dask is unable to
determine the type of output that that function returns.

To resolve this please provide a meta= keyword.
The docstring of the Dask function you ran should have more information.

Original error is below:
------------------------
TypeError('data type not understood',)

Traceback:
---------
  File "/home/nihhaar/workspace/lib/python3.6/site-packages/dask/dataframe/utils.py", line 168, in raise_on_meta_error
    yield
  File "/home/nihhaar/workspace/lib/python3.6/site-packages/dask/dataframe/core.py", line 4725, in _emulate
    return func(*_extract_meta(args, True), **_extract_meta(kwargs, True))
  File "/home/nihhaar/workspace/lib/python3.6/site-packages/dask/dataframe/core.py", line 4705, in _extract_meta
    return tuple([_extract_meta(_x, nonempty) for _x in x])
  File "/home/nihhaar/workspace/lib/python3.6/site-packages/dask/dataframe/core.py", line 4705, in <listcomp>
    return tuple([_extract_meta(_x, nonempty) for _x in x])
  File "/home/nihhaar/workspace/lib/python3.6/site-packages/dask/dataframe/core.py", line 4701, in _extract_meta
    return x._meta_nonempty if nonempty else x._meta
  File "/home/nihhaar/workspace/lib/python3.6/site-packages/dask/dataframe/core.py", line 333, in _meta_nonempty
    return meta_nonempty(self._meta)
  File "/home/nihhaar/workspace/lib/python3.6/site-packages/dask/utils.py", line 505, in __call__
    return meth(arg, *args, **kwargs)
  File "/home/nihhaar/workspace/lib/python3.6/site-packages/dask/dataframe/utils.py", line 385, in meta_nonempty_dataframe
    data = {i: _nonempty_series(x.iloc[:, i], idx=idx) for i, c in enumerate(x.columns)}
  File "/home/nihhaar/workspace/lib/python3.6/site-packages/dask/dataframe/utils.py", line 385, in <dictcomp>
    data = {i: _nonempty_series(x.iloc[:, i], idx=idx) for i, c in enumerate(x.columns)}
  File "/home/nihhaar/workspace/lib/python3.6/site-packages/dask/dataframe/utils.py", line 566, in _nonempty_series
    data = np.array([entry, entry], dtype=dtype)
```

**Comment by jrbourbeau** [comment]
Thanks for reporting and providing a minimal example @Nihhaar! What version of dask are you using? I'm not able to reproduce the error with the latest release (version 2.12.0)

**Comment by Nihhaar** [comment]
Fixed by upgrading to 2.12.0. Should have done it before. Thanks for your time!
